pkgs/development/tools/misc/gf/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib, stdenv, makeWrapper
+
+  ,fetchgit, libX11, pkg-config, gdb
+
+  ,freetypeSupport ? false, freetype
+  ,extensionFolder ? null
+}:
+
+let
+  revision = "e0d6d2f59344f853a4a204d5313db6b6a5e5de7d";
+in
+
+stdenv.mkDerivation rec {
+  pname = "gf";
+  version = "2+${builtins.substring 0 7 revision}";
+
+  src = fetchgit {
+    url = "https://github.com/nakst/gf.git";
+    rev = "${revision}";
+    sha256 = "1b9iggz50l9aksblg1lj64daf2l5kjnkixdw8qjjblp63xz668r7";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config ];
+  buildInputs = [ libX11 gdb ]
+    ++ lib.optional freetypeSupport freetype;
+
+  inherit extensionFolder;
+
+  postPatch = ''
+    if [ ! -z "$extensionFolder" ]; then
+      echo adding extensions from $extensionFolder;
+      cp -r "$extensionFolder"/* .
+    fi
+  '';
+
+  patchPhase = ''
+    echo #!/usr/bin/env bash > build.sh
```

Comment:
What are you trying to do here? This code is messy!

Reply:
fixed

Reply:
Well, I have read the original code, and it is so messy it would be easier to rewrite it instead of circumventing it.

Reply:
I replaced the sed with a patchfile. Or did you mean I should rewite the whole build.sh file?

Reply:
To be honest I prefer to hack the upstream sources, lol

It is OK for now.